packages/core/src/hooks/useEthers.ts

```diff
@@ -1,31 +1,71 @@
-import { useWeb3React } from '@web3-react/core'
-import { Web3Provider } from '@ethersproject/providers'
-import { ChainId } from '../constants'
 import { useCallback } from 'react'
-import { useConfig } from '../providers/config/context'
-import { InjectedConnector } from '@web3-react/injected-connector'
-
-type ActivateBrowserWallet = (onError?: (error: Error) => void, throwErrors?: boolean) => void
+import { JsonRpcProvider } from '@ethersproject/providers'
+import { ChainId } from '../constants'
+import { useInjectedProvider, useNetwork } from '../providers'
 
-export type Web3Ethers = ReturnType<typeof useWeb3React> & {
-  library?: Web3Provider
+export type Web3Ethers = {
+  activate: (provider: JsonRpcProvider) => Promise<void>
+  setError: (error: Error) => void
+  deactivate: () => void
+  connector: undefined
   chainId?: ChainId
-  activateBrowserWallet: ActivateBrowserWallet
+  account?: null | string
+  error?: Error
+  library?: JsonRpcProvider
+  active: boolean
+  activateBrowserWallet: () => void
+}
+
+async function tryToGetAccount(provider: JsonRpcProvider) {
+  try {
+    return await provider.getSigner().getAddress()
+  } catch {
```

Comment:
Would be good to look at exception and find if it is the right one - not to overshadow other exception.
What is the reason why it can be thrown?

Reply:
e.g. distinguish between network error and account not connected